lib/spack/docs/packaging_guide.rst

```diff
@@ -3453,6 +3442,12 @@ directory, including cached virtual indices.
 
 To remove all of the above, the command can be called with ``--all``.
 
+Also, cleans up temporary files for a particular package, by deleting the
+expanded/checked out source code *and* any downloaded archive.  If
+``fetch``, ``stage``, or ``install`` are run again after this, Spack's
+build process will start from scratch.
```

Comment:
I'm not sure if I would expect `spack clean foo` to delete the downloaded archive. Does `spack clean` delete the downloaded archive for everything? If not, we should keep them in line. I think this would be a good setup:
```
$ spack clean --stage foo  # delete stage for foo
$ spack clean foo  # same as above
$ spack clean --downloads foo  # delete downloaded archive for foo
$ spack clean --stage --downloads foo  # both of the above
```
It would actually be really nice if `spack clean --downloads foo` worked. It makes it a lot easier to test fetching for packages I've already downloaded. I've been manually deleting archives for so long.

There's also a reference above that should be updated.

> When called with ``--stage`` or without arguments this removes all staged files and will be equivalent to running ``spack clean`` for every package you have fetched or staged.

This is no longer _equivalent_ to `spack clean`, it _is_ `spack clean`.